### DISCUSSION TITLE
Livewire not working on shared hosting with SSL

### DISCUSSION THREAD

**Discussion by zyruva** [livewire/livewire#4745] [1. Help]
Hello,

I have a Laravel/Livewire website hosted on a shared hosting (OVH, with no access to SSH), and i have set everything up and everything works, but when the website is accessed through SSL, livewire simply doesnt work and it spits out this error on the console:

```
Failed to load resource: net::ERR_HTTP2_PROTOCOL_ERROR livewire.js:1
Uncaught ReferenceError: Livewire is not defined at (index):1329:216
```

I have tried to access https://myapp.com/livewire/livewire.js, and it is accessible, and also remember that this error doesnt show up when using regular HTTP.

My livewire.php has no special settings:
```
'asset_url' => null,
'app_url' => null,
```

and my laravel installation is at the root directory, and public was renamed to www, so im not using a sub-directory inside my www (public) directory

What could be the reason behind this ?

Thank you very much!

**Comment by joshhanley** [comment]
@zyruva sorry, I can't be of much help as I've never used OVH. I only use servers generated using Laravel Forge on like AWS or DigitalOcean.

All my sites use https with Livewire happily. So it must be something to do with the OVH configuration.

**Reply by zyruva** [reply]
Yes i agree, ive tried everything i can to solve it but nothing worked, guess i need to migrate to a better hosting service.
Thank you very much

**Reply by joshhanley** [reply]
@zyruva ah bugger. Yeah sounds like it!

No worries 🙂

**Comment by ades4827** [comment]
Hi, I found these instructions and solved it: https://laracasts.com/discuss/channels/livewire/livewire-https-issue
Also on OVH crappy hosting

update composer.json adding this line
` "@php artisan vendor:publish --force --tag=livewire:assets --ansi"`
in scripts > post-autoload-dump

**Comment by clementlanot** [comment]
Hello, same problem with Livewire 3 for my website in OVH

**Reply by jfvoliveira** [reply]
Hey there. Have you managed to solve this issue?

**Reply by clementlanot** [reply]
I left OVH for infomaniak.

**Comment by Magiczne** [comment]
I've debugged this when trying to deploy site on OVH - the problem is not located in the Livewire itself, but in the Symfony `BinaryResponseFile` - or I should say the issue is that for some stupid reason OVH has very weird problems with writing data to the `php:\\output` which is using in the [`BinaryResponseFile::sendContent`](https://github.com/symfony/http-foundation/blob/6.3/BinaryFileResponse.php#L303) method

The file is accesible when entering URL manually, but it still loads for 5 or so seconsd, so probably the server timeouts loading this file into the standard php output stream.

The "solution", or hack I should say is pretty simple, and pretty ugly but works. I have applied patch to the libary using https://github.com/cweagans/composer-patches:

```diff
--- /dev/null
+++ ../src/Drawer/Utils.php
@@ -73,6 +73,11 @@
             $headers['Content-Encoding'] = 'br';
         }
 
+        return response()->make(
+            content: file_get_contents($file),
+            headers: $headers
+        );
+
         return response()->file($file, $headers);
     }

```

and it works like a charm. Hope someone will find this helpful. But this is definitely not a pretty solution. I'm still trying to get some helpful information from the OVH support, but I'm sceptic about it.

**Comment by Sabrina-Travail-Lewagon** [comment]
Hi, how did you do the patch ? 
Thanks.

**Reply by Magiczne** [reply]
Using https://github.com/symplify/vendor-patches

Just look at the readme, should be pretty self explanatory.

**Comment by gboquizosanchez** [comment]
I just added into .htaccess

```
# Fix HTTP2 Bug on OVH
RewriteRule ^(.*)$ $1 [NS,E=no-gzip:1,E=dont-vary:1]
```
And working like a charm.
